### ISSUE TITLE
Disabling Mission Control animation in Sierra?

### ISSUE THREAD

**Issue by olalonde** [mathiasbynens/dotfiles#711]
It seems that 

```bash
defaults write com.apple.dock expose-animation-duration -float 0.1
```

doesn't take effect in MacOS Sierra.

**Comment by Tatsh** [comment]
Ugh. Is there a user option somewhere, you could diff `defaults read` output before and after?

**Comment by olalonde** [comment]
It writes the setting correctly, but it won't speed up the animation when swiping with 3 fingers to mission control. Used to work before upgrading to Sierra.

**Comment by levithomason** [comment]
I have never wanted to completely destroy my mac in an absolute fit of rage.  Thanks macOS Sierra.

**Comment by levithomason** [comment]
Per [this](https://www.reddit.com/r/MacOS/comments/4s5mjm/macos_sierra_slow_animations_mission_control/) reddit thread:

>Mission Control changed so that you can control the speed with your fingers. Before it went by itself.
>I don't like it, but that's the design.
>You should file a bug for it, since you don't seem to like it either.

Just realized this is true, swiping faster makes the animation faster.  There's got to a fix for this.

**Comment by chdiza** [comment]
In any case, the setting is worth keeping, because AFAICT it *does* help for when one uses the keyboard to go in and out of MC.

**Comment by levithomason** [comment]
In my case, using the keyboard to open MC on macOS Sierra seems to animate at the same speed whether I use:

`defaults write com.apple.dock expose-animation-duration -float 0.1`

or

`defaults write com.apple.dock expose-animation-duration -float 5`

I did `killall Dock` and logged out/in in between testing.

**Comment by essarhash** [comment]
Very simple method with no need for terminal, built into MacOs.

Just enable Reduce Motion in macOS Sierra to Speed Up Mission Control. It removes the Animation.
Settings > Accessibility > Display > Reduce Motion checkbox. 
The animated transition becomes a fade in/fade out.

(Of course this is system wide, not Mission Control Specific. Even the sliding animation when swiping between spaces turns into a fade)

 I also use Reduce Motion on iPhone, its just a preference.

**Comment by Nualiian** [comment]
@essarhash This does not really solve the issue, since the mission control animation still keeps track of your fingers, it just disable the animation.
This is the most annoying thing they ever did to the os, I hate it so much. It also lags sometimes and gets stuck randomly mid-animation 😞

**Comment by essarhash** [comment]
Not sure i understand 'mission control animation still keeps track of your fingers, it just disable the animation’

With accessibility settings, the animations become fades, instead of zoom in and out.

1. Your swiping speed controls the fading speed. So swipe fast, it fades fast. Swipe slow it fades slow — i can imagine this is annoying and can get stuck.
2. However, using the simple keyboard shortcuts for Mission control (in Mission Control  preferences) uses fast fades, and may be not lag. Tried it?

Which  Animation has been changed in Sierra that you’re finding annoying? Sorry if I am missing something in this thread...

**Comment by Johannes-Esseling** [comment]
@essarhash 
I have, as many have pointed out, always set my animation to 0 with
`defaults write com.apple.dock expose-animation-duration -float 0.0`
This would completely turn off animations without annoying fade effects when using mission control.

In MacOS Sierra however this option doesn't seem to affect the duration of the mission control launch. 
Needless to say but I am really disappointed this doesn't work anymore!

**Comment by Nualiian** [comment]
@essarhash 
yep, just like @Johannes-Esseling says, the animation is still dynamic and the option that used to work back before Sierra now just doesn't work and  it seems we are stuck with it being dynamic whether we like it or not

**Comment by gprasant** [comment]
I use keyboard shortcuts to access Mission control. When I do `Ctrl + Up`, it is quicker than the four finger swipe.

**Comment by essarhash** [comment]
You’re right. Thats actually a great shortcut… When i got my mac I just got caught up in the ‘swipe gestures are cool' thing :-)

**Comment by bdefore** [comment]
anyone discover a way to completely disable the transition? even at 0.1 setting the flash of my desktop is jarring

**Comment by Nualiian** [comment]
nothing yet. High Sierra is upon us, the settings might change as they almost always do, so something might be possible then :)

**Comment by ChrisChiera** [comment]
To note, on High Sierra same issue persists. Regardless of the terminal setting, when swiping or CMD + TAB to a full screen window still has a delay versus CMD + Tab to windows on the same desktop. 

Enabling Reduce Motion as others noted makes it fade instead of swipe which appears to be every so slightly faster but still a delay. To note normally notification center swipe out has a delay but the reduce motion feature makes that sidebar appear immediately. So hopefully, eventually they make CMD + Tab full screen items instantly as well.

**Comment by kud** [comment]
If you want to reduce motion via cli:

`defaults write com.apple.universalaccess reduceMotion -bool true`

**Comment by ChrisChiera** [comment]
@Kud Unfortunately that only accomplishes the same affect as what others noted last year when doing, "Settings > Accessibility > Display > Reduce Motion checkbox" which does not solve the issue but rather changes the slide to a fade as discussed above.  Perhaps 10.14 will finally solve the issue. Crossing fingers!

**Comment by kud** [comment]
Yeah I understand. I gave a workaround here because I didn't see the command in this project. :)

**Comment by binamkayastha** [comment]
`defaults write com.apple.dock expose-animation-duration -float 0.1`
Can will have an effect if you use shortcuts like `Command-Left` and `Command-Right`

**Comment by lkulbacki** [comment]
`defaults write com.apple.dock expose-animation-duration -float 0.1` still not working on Mojave. And I thought I could get avoid using external software to amplify my productivity. Ehhh...

**Comment by 1natsu172** [comment]
Solution. 

https://apple.stackexchange.com/questions/253756/speed-up-mission-control-animations-in-macos-sierra

We need BTT but will return to Mac experience. Too speedy.

**Comment by jdanbrown** [comment]
The BTT solution in https://apple.stackexchange.com/questions/253756/speed-up-mission-control-animations-in-macos-sierra (or similarly with TotalSpaces2) solves the switch-desktop delay, but neither solves the Exposé animation delay. I've had to switch to TotalSpaces2's "instant exposé" to eliminate the Exposé animation delay, which unfortunately shows windows across _all_ desktops instead of just the current desktop, but it's the best solution I've found so far...

**Comment by sauron918** [comment]
> Enabling Reduce Motion as others noted makes it fade instead of swipe which appears to be every so slightly faster but still a delay. 

It would be nice to have possibility to remove delays at all 🤔

**Comment by twome** [comment]
*"It just works!"*
Don't you just fucking love closed-source exploitative monopolies like Apple?

**Comment by ssmooncoder** [comment]
Sigh no changes to macOS Catalina either... Is there a non-paid option to remove spaces transition delay?

**Comment by syadnom** [comment]
While the 'reduced motion' option works for me, I would much prefer a very quick animation.   I like the 'movement' to help organize the desktops spatially.

The MacOS experience is function AND beautiful and I enjoy using the system *but* that delay is a productivity hinderance.  I don't have that same hinderance on ubuntu/gnome for example because I can tune it. 

(will still take it as-is over windows 10 though)

**Comment by javivera** [comment]
Fuck you apple, such a simple item and can't be done.
I DONT WANT ANY FUCKING ANIMATIONS SHOVE DOWN MY THROAT

**Comment by javivera** [comment]
Fuck reduced motion, I want to switch workspace in 0.0001 seconds, don't need any animation, im now on a computer for funny animations , im working.. god damn

**Comment by sbusch** [comment]
Yes, its annoying. More than that. But this is the wrong place, please file a bug or something at Apple if you want to be counted, @javivera We're working, too

**Comment by ferrocactus** [comment]
It is possible to disable space switch animations via [Hammerspoon](https://github.com/Hammerspoon/hammerspoon) with the [Spaces API](https://github.com/asmagill/hs._asm.undocumented.spaces). It can be a big buggy at times, but it is good enough for me to use it.

Here is my hammerspoon `init.lua` script for switching spaces: [https://github.com/ferrocactus/hspoonconfig/blob/main/init.lua](https://github.com/ferrocactus/hspoonconfig/blob/main/init.lua).
